types/component.ts

```diff
@@ -0,0 +1,8 @@
+export enum component {
```

Comment:
This shouldn't be a global type. There's not even a need for making an external type at all. This should be defined inline with the component declaration